pkg/server/server.go

```diff
@@ -1002,7 +1002,19 @@ func (server *BgpServer) propagateUpdate(peer *Peer, pathList []*table.Path) {
 					// given RT on RTM NLRI is already removed from adj-RIB-in.
 					_, candidates = server.getBestFromLocal(peer, fs)
 				} else {
-					candidates = server.globalRib.GetBestPathList(peer.TableID(), 0, fs)
+					// https://github.com/osrg/gobgp/issues/1777
+					// Ignore duplicate Membership announcements
+					membershipsForNH := server.globalRib.GetPathListWithNexthop(table.GLOBAL_RIB_NAME, []bgp.RouteFamily{bgp.RF_RTC_UC}, path.GetNexthop())
```

Comment:
Instead of GetPathListWithNexthop(), using GetBestPathList() and comparing Path.GetSource() shows the expected logic more precisely?

Reply:
I believe, limiting searched Paths to only BestPaths may not work as expected, since RR Client may connect to multiple RRs. I'm not sure, whether we should match source or nexthop, though. What do you think?